src/Kestrel.Core/Internal/Http2/Http2Connection.cs

```diff
@@ -89,6 +89,8 @@ public Http2Connection(Http2ConnectionContext context)
             _frameWriter = new Http2FrameWriter(context.Transport.Output, context.ConnectionContext, _outputFlowControl, this, context.ConnectionId, context.ServiceContext.Log);
             _hpackDecoder = new HPackDecoder((int)_serverSettings.HeaderTableSize);
             _serverSettings.MaxConcurrentStreams = (uint)context.ServiceContext.ServerOptions.Limits.Http2.MaxStreamsPerConnection;
+            _serverSettings.MaxFrameSize = (uint)context.ServiceContext.ServerOptions.Limits.Http2.MaxFrameSize;
```

Comment:
I'm not a fan of these conversions between int and uint everywhere. Currently the values on Limits.Http2 are all ints and the values on Http2PeerSettings are uints so we have to convert between the two. We should just pick one and stick to it.